native/src/main/java/io/ballerina/lib/sap/Listener.java

```diff
@@ -102,40 +99,26 @@ public static Object immediateStop(BObject client) {
         return stopListener(client);
     }
 
-    @SuppressWarnings("unchecked")
     public static Object detach(BObject listener, BObject service) {
```

Comment:
Generally the detach pass the service object to be detached since the listener usually supports the multiple services. But in here, regardless of the service passed in the detach method we are removing the one service we are allowing in the listener right?

Reply:
Yes